tools/run_presubmit

```diff
@@ -69,20 +69,40 @@ def run_command(args: List[str],
 
 
 def get_merge_base() -> Optional[str]:
-  """Gets the merge base with the upstream branch (@{u}). Returns None if not
-  found."""
+  result = run_command(["git", "symbolic-ref", "--short", "HEAD"], check=True)
+  branch = result.stdout.strip()
+
+  # First check if there's a tracking branch
   try:
-    run_command(['git', 'rev-parse', '--abbrev-ref', '@{u}'],
-                check=True,
-                stderr=subprocess.DEVNULL)
-    result = run_command(['git', 'merge-base', 'HEAD', '@{u}'])
-    return result.stdout.strip()
+    # More direct way to get the upstream branch
+    args = ["git", "rev-parse", "--abbrev-ref", f"{branch}@{{upstream}}"]
+    tracking_result = run_command(args, check=True)
+    tracking_branch = tracking_result.stdout.strip()
+
+    # If we have a tracking branch, check if it's local
+    if tracking_branch:
+      # Check if this is a local branch by trying to get its ref in refs/heads/
+      try:
+        _ = run_command(
+            ["git", "show-ref", "--verify", f"refs/heads/{tracking_branch}"],
+            check=True)
```

Comment:
add `stderr=subprorcess.DEVNULL` so we don't see confusing errors?

Same below

Reply:
Not necessary, capture_output means output is implicitly captured.